src/test/handlers/graphql.ts

```diff
@@ -0,0 +1,84 @@
+import { suite, test } from "mocha-typescript";
+import { expect } from "chai";
+import { NoDuplicateFields } from "../../handlers/graphql/handler";
+import schema from "../../handlers/graphql/schema";
+import { parse, validate } from "graphql";
+import { specifiedRules } from "graphql/validation";
+
+specifiedRules.push(NoDuplicateFields);
+console.log("initializing specifiedRules: ", specifiedRules);
+
+@suite class GraphqlTest {
+    @test public async "Graphql#validateValidFullSearch()"() {
+        const query = `
+      query Search($query: String!, $last: Int, $before: String) {
+        search(query: $query, last: $last, before: $before) {
+          totalCount
+          pageInfo {
+            hasPreviousPage
+          }
+          edges {
+            cursor
+            node {
+              id
+              action
+              crud
+              created
+              received
+              canonical_time
+              description
+              actor {
+                id
+                name
+                href
+              }
+              group {
+                id
+                name
+              }
+              target {
+                id
+                name
+                href
+                type
+              }
+              display {
+                markdown
+              }
+              is_failure
+              is_anonymous
+              source_ip
+              country
+              loc_subdiv1
+              loc_subdiv2
+            }
+          }
+        }
+      }`;
+
+        var documentAST = parse(query);
+        const errors = validate(schema, documentAST);
+        expect(errors).to.be.empty;
+    }
+
+    @test public async "Graphql#validateDupFields()"() {
+        const query = `
+        query Search($query: String!, $last: Int, $before: String) {
+          search(query: $query, last: $last, before: $before) { totalCount totalCount }
+        }`;
+        var documentAST = parse(query);
+        const errors = validate(schema, documentAST);
+        expect(errors).to.not.be.empty;
+    }
+
+    @test public async "Graphql#validateAliasOverload()"() {
+        const query = `
+        query Search($query: String!, $last: Int, $before: String) {
+          search(query: $query, last: $last, before: $before) { a1:totalCount a2:totalCount }
+        }`;
+        var documentAST = parse(query);
+        const errors = validate(schema, documentAST);
+        expect(errors).to.not.be.empty;
```

Comment:
I would just double check that the errors array contains that Duplicate field error